app/components/concept.ts

```diff
@@ -93,38 +99,66 @@ export default class ConceptComponent extends Component<Signature> {
     }
   }
 
+  get currentBlockGroupIndex() {
+    return this.lastRevealedBlockGroupIndex || 0;
+  }
+
   get visibleBlockGroups() {
     return this.allBlockGroups.slice(0, (this.lastRevealedBlockGroupIndex || 0) + 1);
   }
 
+  findCurrentBlockGroupIndex(completedBlocksCount: number) {
+    let traversedBlocksCount = 0;
+    let currentBlockGroupIndex = 0;
+
+    for (let i = 0; i < this.allBlockGroups.length; i++) {
+      const blockGroup = this.allBlockGroups[i];
+      traversedBlocksCount = traversedBlocksCount + blockGroup!.blocks.length;
+
+      if (traversedBlocksCount > completedBlocksCount) {
+        currentBlockGroupIndex = i;
+        break;
+      } else if (traversedBlocksCount === completedBlocksCount) {
+        currentBlockGroupIndex = i + 1;
+        break;
+      }
+    }
+
+    return currentBlockGroupIndex;
+  }
+
+  enqueueConceptEngagementUpdate = task({ enqueue: true }, async () => {
```

Comment:
@libmartinito how do we choose between `enqueue`, and `keepLatest` here? What are the downsides/upsides to using each?